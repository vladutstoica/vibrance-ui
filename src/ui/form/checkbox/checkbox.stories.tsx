import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Label } from "../label/label";
import { Checkbox } from "./checkbox";

const meta: Meta<typeof Checkbox> = {
  title: "Design System/Form/Checkbox",
  component: Checkbox,
  tags: ["autodocs"],
  argTypes: {
    checked: {
      control: "boolean",
      description: "The controlled checked state of the checkbox",
      table: {
        type: { summary: "boolean" },
      },
    },
    defaultChecked: {
      control: "boolean",
      description: "The default checked state when initially rendered",
      table: {
        type: { summary: "boolean" },
      },
    },
    disabled: {
      control: "boolean",
      description:
        "When true, prevents the user from interacting with the checkbox",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    required: {
      control: "boolean",
      description:
        "When true, indicates that the user must check this checkbox",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Checkbox>;

// Common label styles
const labelStyles =
  "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70";

// Common layout wrapper for checkbox + label combinations
const CheckboxWrapper = ({
  children,
  align = "center",
}: {
  children: React.ReactNode;
  align?: "center" | "top";
}) => <div className={`flex items-${align} space-x-2`}>{children}</div>;

export const Default: Story = {
  decorators: [
    (Story) => (
      <div className="flex items-center space-x-2">
        <Story />
        <Label htmlFor="terms">Accept terms and conditions</Label>
      </div>
    ),
  ],
  args: {
    id: "terms",
  },
};

export const WithText: Story = {
  decorators: [
    (Story) => (
      <CheckboxWrapper align="top">
        <Story />
        <div className="grid gap-1.5 leading-none">
          <Label htmlFor="terms1" className={labelStyles}>
            Accept terms and conditions
          </Label>
          <p className="text-muted-foreground text-sm">
            You agree to our Terms of Service and Privacy Policy.
          </p>
        </div>
      </CheckboxWrapper>
    ),
  ],
  args: {
    id: "terms1",
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    id: "terms-disabled",
  },
  decorators: [
    (Story) => (
      <CheckboxWrapper>
        <Story />
        <Label htmlFor="terms-disabled" className={labelStyles}>
          Accept terms and conditions
        </Label>
      </CheckboxWrapper>
    ),
  ],
};

export const DisabledChecked: Story = {
  args: {
    disabled: true,
    defaultChecked: true,
    id: "terms-disabled-checked",
  },
  decorators: [
    (Story) => (
      <CheckboxWrapper>
        <Story />
        <Label htmlFor="terms-disabled-checked" className={labelStyles}>
          Accept terms and conditions
        </Label>
      </CheckboxWrapper>
    ),
  ],
};

export const Required: Story = {
  args: {
    required: true,
    id: "terms-required",
    "aria-required": true,
  },
  decorators: [
    (Story) => (
      <CheckboxWrapper>
        <Story />
        <Label htmlFor="terms-required" className={labelStyles}>
          Accept terms and conditions{" "}
          <span className="text-red-500" aria-hidden="true">
            *
          </span>
        </Label>
      </CheckboxWrapper>
    ),
  ],
};
