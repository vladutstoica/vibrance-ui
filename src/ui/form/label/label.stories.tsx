import type { Meta, StoryObj } from "@storybook/react";
import * as React from "react";
import { Checkbox } from "../checkbox/checkbox";
import { Input } from "../input/input";
import { Label } from "./label";

const meta: Meta<typeof Label> = {
  title: "Form/Label",
  component: Label,
  tags: ["autodocs"],
  argTypes: {
    htmlFor: {
      description: "The id of the form control this label is associated with",
      control: "text",
    },
  },
};

export default meta;
type Story = StoryObj<typeof Label>;

export const Default: Story = {
  args: {
    children: "Accept terms and conditions",
  },
};

export const WithInput: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="email">Your email address</Label>
      <Input type="email" id="email" placeholder="Email" />
    </div>
  ),
};

export const WithCheckbox: Story = {
  render: () => (
    <div className="flex items-center space-x-2">
      <Checkbox id="terms" />
      <Label htmlFor="terms">Accept terms and conditions</Label>
    </div>
  ),
};

export const Required: Story = {
  render: () => (
    <div className="grid w-full max-w-sm items-center gap-1.5">
      <Label htmlFor="required-email">
        Email <span className="text-red-500">*</span>
      </Label>
      <Input type="email" id="required-email" placeholder="Email" required />
    </div>
  ),
};

export const WithDescription: Story = {
  render: () => (
    <div className="grid w-full max-w-sm gap-1.5">
      <Label htmlFor="picture">Picture</Label>
      <Input id="picture" type="file" />
      <p className="text-muted-foreground text-sm">
        Upload a picture to your profile.
      </p>
    </div>
  ),
};
