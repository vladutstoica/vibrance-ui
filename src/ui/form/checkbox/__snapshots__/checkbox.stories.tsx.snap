// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Form/Checkbox Default smoke-test 1`] = `
<div class="flex items-center space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="false"
          data-state="unchecked"
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms"
  >
  </button>
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="terms"
  >
    Accept terms and conditions
  </label>
</div>
`;

exports[`Design System/Form/Checkbox Disabled smoke-test 1`] = `
<div class="flex items-center space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="false"
          data-state="unchecked"
          data-disabled
          disabled
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms-disabled"
  >
  </button>
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="terms-disabled"
  >
    Accept terms and conditions
  </label>
</div>
`;

exports[`Design System/Form/Checkbox DisabledChecked smoke-test 1`] = `
<div class="flex items-center space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="true"
          data-state="checked"
          data-disabled
          disabled
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms-disabled-checked"
  >
    <span data-state="checked"
          data-disabled
          class="flex items-center justify-center text-current"
          style="pointer-events: none;"
    >
      <svg xmlns="http://www.w3.org/2000/svg"
           width="24"
           height="24"
           viewbox="0 0 24 24"
           fill="none"
           stroke="currentColor"
           stroke-width="2"
           stroke-linecap="round"
           stroke-linejoin="round"
           class="lucide lucide-check h-4 w-4"
      >
        <path d="M20 6 9 17l-5-5">
        </path>
      </svg>
    </span>
  </button>
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="terms-disabled-checked"
  >
    Accept terms and conditions
  </label>
</div>
`;

exports[`Design System/Form/Checkbox Required smoke-test 1`] = `
<div class="flex items-center space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="false"
          aria-required="true"
          data-state="unchecked"
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms-required"
  >
  </button>
  <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
         for="terms-required"
  >
    Accept terms and conditions
    <span class="text-red-500"
          aria-hidden="true"
    >
      *
    </span>
  </label>
</div>
`;

exports[`Design System/Form/Checkbox WithText smoke-test 1`] = `
<div class="flex items-top space-x-2">
  <button type="button"
          role="checkbox"
          aria-checked="false"
          data-state="unchecked"
          value="on"
          class="peer h-4 w-4 shrink-0 rounded-sm border border-primary ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=checked]:text-primary-foreground"
          id="terms1"
  >
  </button>
  <div class="grid gap-1.5 leading-none">
    <label class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
           for="terms1"
    >
      Accept terms and conditions
    </label>
    <p class="text-muted-foreground text-sm">
      You agree to our Terms of Service and Privacy Policy.
    </p>
  </div>
</div>
`;
